pages/developers/mypy.md

```diff
@@ -0,0 +1,169 @@
+   and still requires the proper imports.
+3. They can be in a separate file with the same name but with a `.pyi`
+   extension. This is important for type stubs or for cases where you don't
+   want to add imports or touch the original code. You can annotate compiled
+   files or libraries you don't control this way.
+
+If you have a library you don't control, you can add "type stubs" for it, then
+give MyPy your stubs directory. MyPy will pull the types from your stubs. If
+you are writing code for a Raspberry Pi, for example, you could add the stubs
+for the Pi libraries, and then validate your code, without ever even installing
+the Pi-only libraries!
+
+You do not have to add types for every object - most of the time, you just need
+it for parameters and returns from functions. When running MyPy, you can use
+`reveal_type(...)` to show the inferred type of any object, which is like a
+print statement but at type-checking time, or `reveal_locals()` to see all
+local types.
+
+
+### Configuration
+
+By default, MyPy does as little as possible, so that you can add it iteratively
+to a code base. By default:
+    
+* All untyped variables and return values will be Any
+* Code inside untyped functions is not checked _at all_
+
+You can add configuration to `pyproject.toml` (and a little bit to the files
+themselves), or you can go all the way and pass `--strict`, which will turn on
+everything. Try to turn on as much as possible, and increase it until you can
+run with full `strict` checking. See the [style page][] for configuration
+suggestions.
+
+[style page]: {{ site.baseurl }}{% link pages/developers/style.md %}
+
+For a library to support typing, it has to a) add types using any of the three
+methods, and b) add a `py.typed` empty file to indicate that it's okay to look
+for types inside it. MyPy also looks in `typeshed`, which is a library full of
+type hints for (mostly) the standard library.
+
+Third party libraries that are typed sometimes forget this last step, by the
+way!
+
+### Other features
+
+Static typing has some great features worth checking out:
+
+* Unions (New syntax coming in Python 3.10)
+* Generic Types (New syntax in Python 3.9)
+* Protocols
+* Literals
+* TypedDict
+* Nicer NamedTuple definition (very popular in Python 3 code)
+* MyPy validates the Python version you ask for, regardless of what version you
+  are running.
+
+
+## Complete example
+
+### Python 3.6+
+
+Here's the classic syntax, which you need to use if support 3.6+.
+
+```python
+from typing import Union, List
+
+# Generic types take bracket arguments
+def f(x: int) -> List[int]:
+    return list(range(x))
+
+# Unions are a list of types that all could be allowed
+def g(x: Union[str, int]) -> None:
+    # Type narrowing - Unions get narrowed
+    if isinstance(x, str):
+        print("string", x.lower())
+    else:
+        print("int", x)
+    
+    # Calling x.lower() is invalid here!
+```
+
+Run this with `mypy tmp_mypy1.py --strict` and you'll 
```

Comment:
You forgot to finish this paragraph.

Reply:
```suggestion
```

Actually meant to delete it, as I decided not to show manual runs too often, since we show it as a pre-commit hook mostly.